dynamic_rest/filters.py

```diff
@@ -691,22 +686,66 @@ def remove_invalid_fields(self, queryset, fields, view):
             stripped_term = term.lstrip('-')
             # add back the '-' add the end if necessary
             reverse_sort_term = '' if len(stripped_term) is len(term) else '-'
-            if stripped_term in valid_fields_map:
-                name = reverse_sort_term + valid_fields_map[stripped_term]
-                valid_orderings.append(name)
+            ordering = self.ordering_for(stripped_term, view)
+
+            if ordering:
+                valid_orderings.append(reverse_sort_term + ordering)
             else:
                 invalid_orderings.append(term)
 
         return valid_orderings, invalid_orderings
 
-    def get_valid_fields(self, queryset, view, context={}):
-        """Return valid fields for ordering.
+    def ordering_for(self, term, view):
+        """
+        Return ordering (model field chain) for term (serializer field chain)
+        or None if invalid
 
-        Overwrites DRF's get_valid_fields method so that valid_fields returns
-        serializer fields, not model fields.
+        Raise ImproperlyConfigured if serializer_class not set on view
         """
+        if not self._is_allowed_term(term, view):
```

Comment:
Since we're checking `term` here, that means if `term` is nested we also expect that to appear in the view's `ordering_fields` attribute verbatim. Is that the intention? (i.e. if we want to sort by `foo.bar` or `foo.baz`, or `foo.bam`, `ordering_fields` needs to contain all of those, rather than, say, just `foo`.)

Reply:
I can't speak to the original intention, but I agree that this is a bit weird. I think what is weird about this is attempting to define ordering fields on the view as opposed to the model or the serializer.

So long as `ordering_fields` is supported on the view, I think I am fine with the current implementation. While more verbose, the current implementation affords the ability to block specific nested fields from ordering where just putting `foo` would not.

I am in favor of leaving this as-is for now. In the future, someone could support an addition `ordering_fields` parameter on the serializer or the model to allow for more sustainable ordering fields control.

Thoughts?